modules/engines/engine-hdb/src/main/java/com/sap/xsk/hdb/ds/transformer/hdbdd/HdbddTransformer.java

```diff
@@ -114,8 +126,52 @@ public XSKDataStructureHDBTableModel transformEntitySymbolToTableModel(EntitySym
 
     tableModel.setColumns(tableColumns);
     tableModel.setLocation(location);
-    if (entitySymbol.getAnnotation(CATALOG_ANNOTATION) != null) {
+    if (entitySymbol.getAnnotation(CATALOG_ANNOTATION) != null){
       tableModel.setTableType(entitySymbol.getAnnotation(CATALOG_ANNOTATION).getKeyValuePairs().get(CATALOG_OBJ_TABLE_TYPE).getValue());
+
+      if (entitySymbol.getAnnotation(CATALOG_ANNOTATION).getKeyValuePairs().get(INDEX) != null){
+        List<XSKDataStructureHDBTableIndexModel> indexes = new ArrayList<>();
+        List<XSKDataStructureHDBTableConstraintUniqueModel> uniqueIndexes = new ArrayList<>();
+        AnnotationArray catalogIndexAnnArray = (AnnotationArray) entitySymbol.getAnnotation(CATALOG_ANNOTATION).getKeyValuePairs().get(INDEX);
+
+        for (AbstractAnnotationValue currentAnnValue : catalogIndexAnnArray.getValues()){
+          AnnotationObj annObject = (AnnotationObj) currentAnnValue;
+          Boolean isUnique = annObject.getValue(UNIQUE) != null && Boolean.parseBoolean(annObject.getValue(UNIQUE).getValue());
+          String name = annObject.getValue(NAME) != null ? annObject.getValue(NAME).getValue() : null;
+          String order = annObject.getValue(ORDER) != null ? annObject.getValue(ORDER).getValue() : null;
+          Set<String> indexColumnSet = new HashSet<>();
+
+          ((AnnotationArray) annObject.getValue(ELEMENT_NAMES)).getValues().forEach(currentElement -> {
+            indexColumnSet.add(currentElement.getValue());
+          });
+
+          if (!isUnique){
+            XSKDataStructureHDBTableIndexModel indexModel = new XSKDataStructureHDBTableIndexModel();
```

Comment:
Couldn't this model's constructor be changed in order to accept needed parameters? Having mutable objects with setter methods leads to a harder understanding of which parameters are necessary for creating the object and it may also lead to not knowing when is an object actually created